pkg/cluster/internal/create/actions/installstorage/storage.go

```diff
@@ -53,7 +55,7 @@ func (a *action) Execute(ctx *actions.ActionContext) error {
 	node := controlPlanes[0] // kind expects at least one always
 
 	// add the default storage class
-	if err := addDefaultStorageClass(node); err != nil {
+	if err := addDefaultStorage(ctx.Logger, node); err != nil {
```

Comment:
should we be able to opt-out and not install it as we did with the CNI?
or opt-in and keep the legacy by default?

Reply:
Not necessary initially.
You can easily replace the storage class with your own and delete the deployment, unlike CNI.

Let me mark this PR WIP.

Reply:
There's no point in opt-in. We ship hermetic images, the images etc. will already be there and just running the deployment is low cost.

And the legacy one is know to have major limitations.